codec_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+import (
+	"testing"
+
+	"github.com/bufbuild/connect-go/internal/assert"
+	pingv1 "github.com/bufbuild/connect-go/internal/gen/connect/ping/v1"
+)
+
+func TestCodec(t *testing.T) {
+	t.Parallel()
+	assertCodecRoundTrips(t, &protoBinaryCodec{})
+	assertCodecRoundTrips(t, &protoJSONCodec{})
+	assertStableMarshalEquals(
+		t,
+		&protoBinaryCodec{},
+		&pingv1.PingRequest{Text: "text", Number: 1}, []byte{
+			// Comments are in the format of protoscope.
+			// 1:VARINT 1
+			0b00001_000, 1,
+			// 2:LEN 4 "test"
+			0b00010_010, 4, 0x74, 0x65, 0x78, 0x74,
+		},
+	)
+	assertStableMarshalEquals(
+		t,
+		&protoJSONCodec{},
+		&pingv1.PingRequest{Text: "text", Number: 1},
+		[]byte(`{"number":"1","text":"text"}`),
+	)
+}
```

Comment:
A few suggestions to improve test coverage:

* I always like `testing/quick` to do these sorts of round-trip testing. There are a handful of examples in this repository.
* Rather than directly asserting the wire format, can we create a protobuf message with a `map[string]string` and assert that repeatedly marshaling it always produces the same output?

Reply:
It was a little tricky but I think I have made tests that work how you wanted.